includes/binary_object_upload.form.inc

```diff
@@ -93,27 +93,26 @@ function islandora_binary_object_upload_form_submit(array $form, array &$form_st
   module_load_include('inc', 'islandora', 'includes/utilities');
   $object = islandora_ingest_form_get_object($form_state);
 
-  if (!isset($object['OBJ'])) {
-    $datastream = $object->constructDatastream('OBJ', 'M');
-  }
-  else {
-    $datastream = $object['OBJ'];
-  }
-  $file = file_load($form_state['values']['file']);
-  $datastream->setContentFromFile($file->uri, FALSE);
-  if ($datastream->label != $file->filename) {
-    $datastream->label = $file->filename;
-  }
-  if ($datastream->mimetype != $file->filemime) {
-    $datastream->mimetype = $file->filemime;
-  }
+  if (isset($form_state['values']['file'])) {
+    $file = file_load($form_state['values']['file']);
+    $datastream->setContentFromFile($file->uri, FALSE);
```

Comment:
Looks like [my comment](https://github.com/Islandora-Labs/islandora_binary_object/pull/26#discussion_r128288543) was somewhat misunderstood (or incompletely implemented)...`$datastream` is no longer set here, so it attempts to call `setContentFromFile` on something which is not a datastream object:

```
Fatal error: Call to a member function setContentFromFile() on a non-object in /var/www/drupal7/sites/all/modules/islandora_binary_object/includes/binary_object_upload.form.inc on line 98
[...]
```

Still needs one of the if/else or the ternary/condition operator (with the question mark and and colon `condition ? value if true : value if false`) bit before it attempts to manipulate `$datastream`.